From 95de7dd265f6065bc7d9b5f9689d0c7365c5ba1e Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Sat, 22 Nov 2025 20:03:56 +0100
Subject: [PATCH] Fix extension of placo python module on Windows

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8635233d..f6c4afc6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,6 +157,9 @@ add_library(placo MODULE
 )
 set_target_properties(placo PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${PYTHON_SITELIB})
 target_compile_definitions(placo PUBLIC -DBOOST_DISABLE_PRAGMA_MESSAGE)
+if(WIN32)
+set_target_properties(placo PROPERTIES SUFFIX ".pyd")
+endif()
 
 target_link_libraries(placo ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} libplaco eigenpy::eigenpy)
 target_include_directories(placo PRIVATE ${PYTHON_INCLUDE_DIRS})
