diff --git a/python/pyproject.toml b/python/pyproject.toml
index e69c25c1a..c203fa12e 100644
--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -13,7 +13,7 @@ name              = 'dismod_at'
 version           = '2026.0.1'
 description       = 'Disease Rates as Functions of Age and Time'
 readme            = 'README.md'
-requires-python   = '>=3.0'
+requires-python   = '>=3.10'
 keywords          = [ 'public health', 'disease', 'cause of death' ]
 authors           = [
   {name  = 'Bradley M. Bell', email = 'bradbell@seanet.com'},
@@ -33,7 +33,8 @@ classifiers       = [
 ]
 #
 # https://github.com/pypa/setuptools/issues/4903
-license           = { text = 'AGPL-3.0-or-later' }
+# license           = { text = 'AGPL-3.0-or-later' }
+license = 'AGPL-3.0-or-later'
 #
 # -----------------------------------------------------------------------------
 [tool.setuptools]
