From 29d490ad7a42c38633752c9c6f234ee980421801 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Wed, 24 Apr 2024 19:01:06 -0500
Subject: [PATCH] Update pyproject.toml

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index f5cad29b..1d5be956 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,6 @@ version = "v1.7.4"
 license = {file = "LICENSE"}
 readme = "README.md"
 dependencies = [
-    "openbabel-wheel",
     "numpy",
     "scipy",
     "pandas",
