From e12a3e4d35172af0f811baa152729372c5b31777 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 30 Sep 2024 14:57:28 -0500
Subject: [PATCH] Fix for freethreading build

---
 crossenv/__init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crossenv/__init__.py b/crossenv/__init__.py
index 76dcd5a..0d0602f 100644
--- a/crossenv/__init__.py
+++ b/crossenv/__init__.py
@@ -276,7 +276,8 @@ def find_host_python(self, host):
             sysconfig_paths = [os.path.join(self.host_project_base, build_dir)]
         else:
             self.host_home = self.find_installed_host_home()
-            python_ver = 'python' + sysconfig.get_config_var('py_version_short')
+            py_thread = 't' if sysconfig.get_config_var('Py_GIL_DISABLED') else ''
+            python_ver = 'python' + sysconfig.get_config_var('py_version_short') + py_thread
             libdir = os.path.join(self.host_home, 'lib', python_ver)
             sysconfig_paths = [
                 libdir,
