--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,6 @@
 # Needed for test builds in windows
- include_directories("C:\\Users\\user\\AppData\\Local\\miniforge3\\envs\\gds-sigp\\Library\\include")
+# include_directories("C:\\Users\\user\\AppData\\Local\\miniforge3\\envs\\gds-sigp\\Library\\include")
+include_directories("$ENV{PREFIX}\\include" "$ENV{PREFIX}\\Library\\include")
 
 add_subdirectory(Base)
 add_subdirectory(config)
