diff --git a/at_cascade/csv/get_header.py b/at_cascade/csv/get_header.py
index 59f7cea..25ffe45 100644
--- a/at_cascade/csv/get_header.py
+++ b/at_cascade/csv/get_header.py
@@ -3,7 +3,7 @@
 # SPDX-FileContributor: 2021-25 Bradley M. Bell
 # ----------------------------------------------------------------------------
 import csv
-"""
+r"""
 {xrst_begin csv.get_header}
 
 Get the Header from a CSV File
