diff --git a/build_lib.py b/build_lib.py
index f788b695..001a504c 100644
--- a/build_lib.py
+++ b/build_lib.py
@@ -382,6 +382,8 @@ def main(argv: list[str] | None = None) -> int:
             if not args.host_compiler:
                 print("Warp build error: Could not find MSVC compiler")
                 return 1
+    else:
+        args.host_compiler = build_dll.find_host_compiler()

     try:
         # Handle CI nightly builds (returns updated version string if triggered, else None)
diff --git a/warp/_src/build_dll.py b/warp/_src/build_dll.py
index 335f2872..b5319023 100644
--- a/warp/_src/build_dll.py
+++ b/warp/_src/build_dll.py
@@ -124,7 +124,7 @@ def find_host_compiler():

     else:
         # try and find g++
-        return run_cmd("which g++").decode()
+        return run_cmd("echo $CXX").decode().strip() if run_cmd("echo $CXX") != b"\xA0" else run_cmd("which g++").decode().strip()


 def get_cuda_toolkit_version(cuda_home) -> tuple[int, int]:
@@ -514,7 +514,7 @@ def build_dll_for_arch(args, dll_path, cpp_paths, cu_paths, arch, libs: list[str
     else:
         # Unix compilation
         cuda_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else "nvcc"
-        cpp_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else "g++"
+        cpp_compiler = "clang++" if getattr(args, "clang_build_toolchain", False) else args.host_compiler

         cpp_includes = f' -I"{warp_home_path.parent}/external/llvm-project/out/install/{mode}-{arch}/include"'
         cpp_includes += f' -I"{warp_home_path.parent}/_build/host-deps/llvm-project/release-{arch}/include"'
