diff --git a/warp/_src/build_dll.py b/warp/_src/build_dll.py
index 335f2872..dba1e1a6 100644
--- a/warp/_src/build_dll.py
+++ b/warp/_src/build_dll.py
@@ -98,7 +98,7 @@ def find_host_compiler():
         vs_path = run_cmd(f'"{vswhere_path}" -latest -property installationPath').decode().rstrip()
         vsvars_path = os.path.join(vs_path, "VC\\Auxiliary\\Build\\vcvars64.bat")

-        output = run_cmd(f'"{vsvars_path}" && set').decode()
+        output = run_cmd(f'set').decode()

         for line in output.splitlines():
             pair = line.split("=", 1)
