diff --git a/build_llvm.py b/build_llvm.py
index 1aa8631a..3f1d11f8 100644
--- a/build_llvm.py
+++ b/build_llvm.py
@@ -76,10 +76,6 @@ def build_llvm_clang_from_source_for_arch(args, arch: str, llvm_source: str) ->

     # Check out the LLVM project Git repository, unless it already exists
     if not os.path.exists(llvm_source):
-        # Install dependencies
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "gitpython"])
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "cmake"])
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "ninja"])

         from git import Repo
