From 40247fd7ea3728402c07d361a21a4ea472019a45 Mon Sep 17 00:00:00 2001
From: HeavyTony2 <97621927+HeavyTony2@users.noreply.github.com>
Date: Fri, 2 Jan 2026 14:07:26 -0500
Subject: [PATCH] Update v8_build.py

---
 builder/v8_build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builder/v8_build.py b/builder/v8_build.py
index 4800ac5..090bb8c 100644
--- a/builder/v8_build.py
+++ b/builder/v8_build.py
@@ -24,7 +24,7 @@ basicConfig()
 LOGGER = getLogger(__name__)
 LOGGER.setLevel(DEBUG)
 ROOT_DIR = Path(__file__).absolute().parents[1]
-V8_VERSION = "branch-heads/14.3"
+V8_VERSION = "branch-heads/14.0"
 
 
 @cache
-- 
2.34.1.windows.1

