From 39ed8aec98a42cf56c28537c88b743d34d091ca2 Mon Sep 17 00:00:00 2001
From: gvarnavi <gvarnavides@berkeley.edu>
Date: Wed, 26 Nov 2025 22:34:55 +0100
Subject: [PATCH] removing torchsummary

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 6f4e6c2..9223ae7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,6 @@ dependencies = [
   "torchvision>=0.22.1",
   "tensorboard>=2.19.0",
   "torchmetrics>=1.7.3",
-  "torchsummary>=1.5.1",
   "optuna>=4.5.0",
   "hdf5plugin>=6.0.0",
 ]
-- 
2.50.1 (Apple Git-155)

