From afd2bf669875adadf18451b3bcc9162b1d05ebdd Mon Sep 17 00:00:00 2001
From: gvarnavi <gvarnavides@berkeley.edu>
Date: Wed, 26 Nov 2025 22:11:12 +0100
Subject: [PATCH] removing torchsummary

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index efe5d62..2538572 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,6 @@ dependencies = [
   "torchvision>=0.22.1",
   "tensorboard>=2.19.0",
   "torchmetrics>=1.7.3",
-  "torchsummary>=1.5.1",
   "optuna>=4.5.0",
 ]
 
-- 
2.50.1 (Apple Git-155)

