From 4783e442a76315b62a1f2e683076b9bf580b44ae Mon Sep 17 00:00:00 2001
From: Sherman Siu <s8siu@uwaterloo.ca>
Date: Sun, 31 Aug 2025 20:37:56 -0400
Subject: [PATCH 5/7] Configure build to target aarch64 even though CMake finds
 x86_64 hardware

---
 cmake/cpu_extension.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/cpu_extension.cmake b/cmake/cpu_extension.cmake
index 064944632..5d66767d7 100644
--- a/cmake/cpu_extension.cmake
+++ b/cmake/cpu_extension.cmake
@@ -103,6 +103,12 @@ else()
     find_isa(${CPUINFO} "S390" S390_FOUND)
 endif()
 
+
+set(ASIMD_FOUND TRUE)
+set(AVX512_FOUND FALSE)
+set(AVX2_FOUND FALSE)
+
+
 if (AVX512_FOUND AND NOT AVX512_DISABLED)
     list(APPEND CXX_COMPILE_FLAGS
         "-mavx512f"
