From 8a78b3449af3b75b4ab5dbe0fc8b29958a39d51e Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 9 Jan 2026 10:52:21 +1100
Subject: [PATCH 7/7] pass required CUTLASS_NVCC_ARCHS through to cutlass

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ad89d702..fce9914cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -284,6 +284,9 @@ if(VLLM_GPU_LANG STREQUAL "CUDA")
         GIT_SHALLOW TRUE
     )
   endif()
+  if(DEFINED ENV{CUTLASS_NVCC_ARCHS})
+    set(CUTLASS_NVCC_ARCHS "$ENV{CUTLASS_NVCC_ARCHS}" CACHE STRING "CUDA architectures for CUTLASS")
+  endif()
   FetchContent_MakeAvailable(cutlass)
 
   list(APPEND VLLM_EXT_SRC
