From c15e3ce709eb30017bfa346719d1bef7d157cd27 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Mon, 29 Sep 2025 22:04:46 +0200
Subject: [PATCH] numpy

---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 27b8519..2434b26 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,7 +24,7 @@ classifiers = [
     "Programming Language :: Python :: 3.13",
 ]
 dependencies = [
-    "numpy>=1.23.5,<=2.3.2",
+    "numpy>=1.23.5,<=2.3.3",
     "pint>=0.18,<=0.25",
 ]
 dynamic = ["version"]
@@ -36,7 +36,7 @@ Repository = "https://github.com/pyiron/pyiron_dataclasses"
 
 [project.optional-dependencies]
 convert = [
-    "h5io_browser>=0.1.2,<=0.2.1",
+    "h5io_browser>=0.1.2,<=0.2.2",
 ]
 
 [tool.ruff]
