From 43246541bb73320eee4789213d59aa2276cfe0da Mon Sep 17 00:00:00 2001
From: Lindsey Gray <lindsey.gray@gmail.com>
Date: Thu, 19 Jun 2025 19:46:57 -0500
Subject: [PATCH] use imported getopt

---
 examples/CMakeLists.txt | 27 +++++++++++++++++++--------
 1 file changed, 19 insertions(+), 8 deletions(-)

diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 0b26cda..98ba86c 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -26,7 +26,11 @@ if (MSVC)
   # This lets us have linux-like getopt on Windows.
   # It is not marked required so that it's less difficult
   # to inject the package into cmake via other means.
-  find_package(unofficial-getopt-win32)
+  add_library(getopt SHARED IMPORTED)
+  set_target_properties(getopt PROPERTIES
+                          IMPORTED_IMPLIB "$ENV{PREFIX}\\Library\\lib\\getopt.lib"
+                          IMPORTED_LOCATION "$ENV{PREFIX}\\Library\\bin\\getopt.dll"
+                       )
   
   # The following defines silence warnings about sscanf and other
   # c functions that don't have buffer-overrun protection built into them.
@@ -40,10 +44,17 @@ if (MSVC)
   target_compile_definitions(sample PUBLIC -D_CRT_SECURE_NO_WARNINGS)
   target_compile_definitions(spherical PUBLIC -D_CRT_SECURE_NO_WARNINGS)
 
-  target_link_libraries(siscone_main PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(siscone_area PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(siscone_test PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(times PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(sample PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(spherical PRIVATE unofficial::getopt-win32::getopt)
-endif()
\ No newline at end of file
+  target_include_directories(siscone_main PRIVATE "$ENV{PREFIX}\\Library\\include")
+  target_include_directories(siscone_area PRIVATE "$ENV{PREFIX}\\Library\\include")
+  target_include_directories(siscone_test PRIVATE "$ENV{PREFIX}\\Library\\include")
+  target_include_directories(times PRIVATE "$ENV{PREFIX}\\Library\\include")
+  target_include_directories(sample PRIVATE "$ENV{PREFIX}\\Library\\include")
+  target_include_directories(spherical PRIVATE "$ENV{PREFIX}\\Library\\include")
+
+  target_link_libraries(siscone_main PRIVATE getopt)
+  target_link_libraries(siscone_area PRIVATE getopt)
+  target_link_libraries(siscone_test PRIVATE getopt)
+  target_link_libraries(times PRIVATE getopt)
+  target_link_libraries(sample PRIVATE getopt)
+  target_link_libraries(spherical PRIVATE getopt)
+endif()
-- 
2.39.5 (Apple Git-154)
