From e36e8adec5c17f9931116beab9bf94e588975740 Mon Sep 17 00:00:00 2001
From: Andrew Robbins <andrew@robbinsa.me>
Date: Mon, 12 Jan 2026 18:21:48 -0500
Subject: [PATCH] Correct non-pkgconf case for HWLOC

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0035f7..7412866 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -325,7 +325,11 @@ add_library(hw_detect OBJECT ${PROJECT_SOURCE_DIR}/common/hw_detect.c)
 find_package(HWLOC)
 
 if (HWLOC_FOUND)
-    target_link_libraries(hw_detect PUBLIC "$<IF:$<BOOL:pkgcfg_lib_HWLOC_hwloc>,${pkgcfg_lib_HWLOC_hwloc},${HWLOC_LDFLAGS}>")
+    if(pkgcfg_lib_HWLOC_hwloc)
+         target_link_libraries(hw_detect PUBLIC ${pkgcfg_lib_HWLOC_hwloc})
+    else()
+         target_link_libraries(hw_detect PUBLIC ${HWLOC_hwloc_LIBRARY})
+    endif()
     target_include_directories(hw_detect PRIVATE "${HWLOC_INCLUDE_DIRS}")
 elseif (MSVC)
     FetchContent_Declare(hwloc
-- 
2.52.0

