From 765b099a710427f8eb8e1a0bbab882414a02999c Mon Sep 17 00:00:00 2001
From: Andrew Robbins <andrew@robbinsa.me>
Date: Mon, 7 Jul 2025 14:30:57 -0400
Subject: [PATCH] correct numpy version constraints

---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 03a15250..68861274 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@ maintainers = [
 ]
 
 dependencies = [
-    "numpy>=1.23.5, <2.3",
+    "numpy>=2.0, <3.0",
     "threadpoolctl>=3.4.0, <4.0",
     "scipy>=1.12.0",
     "progressbar2>=4.5.0",
@@ -53,7 +53,7 @@ ]
 
 [build-system]
-requires = ["scikit-build-core>=0.10", "nanobind>=2.7", "numpy>=2.0"]
+requires = ["scikit-build-core>=0.10", "nanobind>=2.7", "numpy>=2.0, <3.0"]
 build-backend = "scikit_build_core.build"
 
 [tool]
