From 4042164181bc50306cfe810af6d7b3b55a411e00 Mon Sep 17 00:00:00 2001
From: Duncan Macleod <macleoddm@cardiff.ac.uk>
Date: Wed, 17 Dec 2025 11:50:25 +0000
Subject: [PATCH] Use IS_WRITABLE unary test

Requires cmake 3.29.0
---
 CMakeLists.txt | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 474dc22..1cdccee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.29.0 FATAL_ERROR)
 project(
     pyfd
     LANGUAGES C
@@ -23,18 +23,11 @@ find_package(Python3 REQUIRED)
 set(PYFD_DEPENDENCIES "fd")
 pkg_check_modules(PYFDDEPS REQUIRED IMPORTED_TARGET ${PYFD_DEPENDENCIES})
 
-#Check if site-packages location is writable
-execute_process(
-        COMMAND "test -w ${Python3_SITEARCH}"
-        RESULT_VARIABLE writable)
-
-#If writable then put it in the site-packages directory otherwise put it in 
-if(${writable} EQUAL 0)
+if(IS_WRITABLE ${Python3_SITEARCH})
     set(TARGET_SP_DIR ${Python3_SITEARCH})
     set(TARGET_INSTALL_DIR ${Python3_SITEARCH}/PyFd)
 else()
-
-    #If not writabale then create a python lib directory from the CMAKE_INSTALL_PREFIX
+    # If not writable then create a python lib directory from the CMAKE_INSTALL_PREFIX
     execute_process(
         COMMAND "${Python3_EXECUTABLE}" "-c"
         "from distutils import sysconfig;print(sysconfig.get_python_lib(plat_specific=True, prefix=''))"
-- 
2.39.5

