Subject: [PATCH] remove numpy init wrap
---
Index: bindings/python/tfel/InitNumpy.cxx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bindings/python/tfel/InitNumpy.cxx b/bindings/python/tfel/InitNumpy.cxx
--- a/bindings/python/tfel/InitNumpy.cxx	(revision 85554f233306548d8c9c4d36af54b715c608b8c7)
+++ b/bindings/python/tfel/InitNumpy.cxx	(date 1738571158615)
@@ -15,21 +15,11 @@
 #include <boost/python/numpy.hpp>
 #include "TFEL/Numpy/InitNumpy.hxx"
 
-#if PY_MAJOR_VERSION == 2
-static void wrapInitializeNumPy() { import_array(); }
-#else
-static void* wrapInitializeNumPy() {
-  import_array();
-  return nullptr;
-}
-#endif
-
 namespace tfel::numpy {
 
   void initializeNumPy() {
     Py_Initialize();
     boost::python::numpy::initialize();
-    ::wrapInitializeNumPy();
   }  // end of initializeNumPy
 
 }  // end of namespace tfel::numpy
