From 38e71f96546a217ea97cd4891b001f6da6f002db Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Fri, 21 Nov 2025 08:30:18 -0700
Subject: [PATCH] fix: Support longer strings for filenames

---
 src/SPheno4.f90 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/SPheno4.f90 b/src/SPheno4.f90
index c63ce27..260f562 100644
--- a/src/SPheno4.f90
+++ b/src/SPheno4.f90
@@ -72,7 +72,7 @@ Implicit None
  !------------------------------------------------------
  ! Input and output files
  !------------------------------------------------------
-  Character(len=120) :: inputFileName, outputFileName
+  Character(len=200) :: inputFileName, outputFileName
 
  Iname = 1
  NameOfUnit(1) = "SPheno4"
-- 
2.52.0

