From 102f4629638acea12165ddbac972fbc208f85e6e Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Thu, 10 Oct 2024 09:38:57 +1000
Subject: [PATCH 3/5] Apply CMAKE_ARGS if set

---
 tools/setup_helpers/extension.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/setup_helpers/extension.py b/tools/setup_helpers/extension.py
index 8825d7d8..dea97335 100644
--- a/tools/setup_helpers/extension.py
+++ b/tools/setup_helpers/extension.py
@@ -3,6 +3,7 @@ import os
 import platform
 import subprocess
 from pathlib import Path
+import shlex
 
 import torch
 from setuptools import Extension
@@ -111,6 +112,9 @@ class CMakeBuild(build_ext):
             f"-DUSE_CUDA:BOOL={'ON' if _USE_CUDA else 'OFF'}",
             f"-DUSE_OPENMP:BOOL={'ON' if _USE_OPENMP else 'OFF'}",
         ]
+        if "CMAKE_ARGS" in os.environ:
+            cmake_args += shlex.split(os.environ["CMAKE_ARGS"])
+
         build_args = ["--target", "install"]
         # Pass CUDA architecture to cmake
         if _TORCH_CUDA_ARCH_LIST is not None:
