Subject: [PATCH] site packages override on windows
---
Index: bindings/python/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
--- a/bindings/python/CMakeLists.txt	(revision 12b4e3217d0b1bac7949a5c1d3b834cd134d883d)
+++ b/bindings/python/CMakeLists.txt	(date 1710603826496)
@@ -1,18 +1,10 @@
 # store the location of the python modules in the MGIS_PYTHON_MODULES_INSTALL_DIRECTORY variable
-if(MGIS_APPEND_SUFFIX)
-  if(WIN32)
-    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
-        bin/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis_${MGIS_SUFFIX_FOR_PYTHON_MODULES})
-  else(WIN32)
-    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
-        "lib${LIB_SUFFIX}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis_${MGIS_SUFFIX_FOR_PYTHON_MODULES}")
-  endif(WIN32)
-else(MGIS_APPEND_SUFFIX)
-  if(WIN32)
-    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
-        "bin/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis")
-  else(WIN32)
-    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
-        "lib${LIB_SUFFIX}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis")
-  endif(WIN32)
-endif(MGIS_APPEND_SUFFIX)
+if(NOT MGIS_PYTHON_MODULES_INSTALL_DIRECTORY)
+  if(MGIS_APPEND_SUFFIX)
+    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
+        "lib${LIB_SUFFIX}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis_${MGIS_SUFFIX_FOR_PYTHON_MODULES}")
+  else(MGIS_APPEND_SUFFIX)
+    set(MGIS_PYTHON_MODULES_INSTALL_DIRECTORY
+        "lib${LIB_SUFFIX}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/mgis")
+  endif(MGIS_APPEND_SUFFIX)
+endif()
