--- libpng-1.6.49.orig/CMakeLists.txt	2025-06-12 14:05:46.000000000 -0300
+++ libpng-1.6.49/CMakeLists.txt	2025-06-16 13:40:19.280938475 -0300
@@ -1022,7 +1022,11 @@
   set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
   set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
   set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
-  set(LIBS "-lz -lm")
+  if(NOT "${M_LIBRARY}" STREQUAL "")
+    set(LIBS        "-lz -lm")
+  else()
+    set(LIBS        "-lz")
+  endif()
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in"
                  "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
                  @ONLY)
