From 8131394255b5459be980ba677c295c54c9f24467 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 22 Nov 2024 18:18:05 +0100
Subject: [PATCH] Update setup.py

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index b4212cb9d..21fe0cc0a 100644
--- a/setup.py
+++ b/setup.py
@@ -113,7 +113,7 @@ def run(self) -> None:
         super().run()
         os.makedirs("driver", exist_ok=True)
         os.makedirs("playwright/driver", exist_ok=True)
-        self._download_and_extract_local_driver()
+        # self._download_and_extract_local_driver()
 
         wheel = None
         if os.getenv("PLAYWRIGHT_TARGET_WHEEL", None):
@@ -133,7 +133,7 @@ def run(self) -> None:
                 )
             )[0]
         assert wheel
-        self._build_wheel(wheel)
+        # self._build_wheel(wheel)
 
     def _build_wheel(
         self,
