--- a/configure
+++ b/configure
@@ -22347,8 +22347,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu

         if test "$cross_compiling" = yes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: test program execution failed" >&5
-$as_echo "$as_me: WARNING: test program execution failed" >&2;}
+  ax_cv_libuuid_is_required=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -22467,8 +22466,7 @@ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext
 ac_compiler_gnu=$ac_cv_c_compiler_gnu

           if test "$cross_compiling" = yes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: test program execution failed" >&5
-$as_echo "$as_me: WARNING: test program execution failed" >&2;}
+  ax_cv_libuuid_works=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -22680,8 +22678,7 @@ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext
 ac_compiler_gnu=$ac_cv_c_compiler_gnu

         if test "$cross_compiling" = yes; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: test program execution failed" >&5
-$as_echo "$as_me: WARNING: test program execution failed" >&2;}
+  ax_cv_uuid_generate_time_safe=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
